src/builtins.ts

```diff
@@ -3587,8 +3588,9 @@ function builtin_unchecked(ctx: BuiltinContext): ExpressionRef {
     checkArgsRequired(ctx, 1)
   ) return module.unreachable();
   let flow = compiler.currentFlow;
-  let alreadyUnchecked = flow.is(FlowFlags.UncheckedContext);
-  flow.set(FlowFlags.UncheckedContext);
+  let ignoreUnchecked = compiler.options.uncheckedBehavior === UncheckedBehavior.Never;
+  let alreadyUnchecked = !ignoreUnchecked && flow.is(FlowFlags.UncheckedContext);
+  if (!ignoreUnchecked) flow.set(FlowFlags.UncheckedContext);
```

Comment:
Would an `assert` make sense here, since these are mutually exclusive?
```suggestion
  let ignoreUnchecked = compiler.options.uncheckedBehavior === UncheckedBehavior.Never;
  let alreadyUnchecked = flow.is(FlowFlags.UncheckedContext);
  if (ignoreUnchecked) assert(!alreadyUnchecked);
  else flow.set(FlowFlags.UncheckedContext);
```